File: hello/lib/hello.py
import sys
import getpass

#author: Aashir Tuladhar (tuladhar@123.com)

#default for no arg
name = "World"

# adding in a class to get user name
def loggedUser():
    return getpass.getuser()

#set name var for arg passed in
if(len(sys.argv)>=2):
    name = sys.argv[1]

#print
print("Hello {0}".format(loggedUser()))

File: hello/lib/hello.c
#include <stdio.h>

int main(int argc, char *argv[])
{
	if (argc < 2)
	{
		printf("Usage: %s <-String to be printed out->\n", argv[0]);
	}

	printf("Hello %s!\n", argv[1]);
	return 0;
}
